docs/design/core/metrics/MetricsList.md

```diff
@@ -0,0 +1,128 @@
+Here is the detailed list of metrics that SDK can collect. Each metric belongs to a category. If a category is enabled, then all metrics belonging to that category will be collected by the SDK.
+
+## Category
+
+1) Default - All metrics under this category will be collected when the metrics are enabled
+2) HttpClient - Additional information collected for http client. The metrics collected for each http client can vary
+3) All - All metrics collected by the SDK comes under this category. This can be useful for debugging purposes.
+
+Note: When metrics feature is enabled, only the `Default` category metrics are collected. Other categories should be explicitly enabled.
+
+## Information collected at application level (Category: Default)
+
+| Metric Name                    | Meter         | Description      |
+| ------------------             | -----------   | ---------------- |
+| RequestCount                   | Counter       | Total number of requests (successful and failed) made from your code to AWS services
+| SuccessRequestCount            | Counter       | Total number of requests from your code to AWS services that resulted in a successful response
+| FailedRequestCount             | Counter       | Total number of requests from your code to AWS services that resulted in a failure. This can be expanded later to categorize the failures into buckets (like ClientErrorCount, ServiceErrorCount, ConnectionErrorCount etc)
+                  
+## Information collected for each request (ApiCall) (Category: Default)
+
+| Metric Name                    | Meter         | Description      |
+| ------------------             | -----------   | ---------------- |
+| Service                        | ConstantGauge | Service ID of the AWS service that the API request is made against
+| Api                            | ConstantGauge | The name of the AWS API the request is made to
+| StreamingRequest               | ConstantGauge | True if the request has streaming payload
+| StreamingResponse              | ConstantGauge | True if the response has streaming payload
+| ApiCallStartTime               | Timer         | The start time of the request
+| ApiCallEndTime                 | Timer         | The end time of the request
+| ApiCallLatency                 | Timer         | The total time taken to finish a request (inclusive of all retries), ApiCallEndTime - ApiCallStartTime
+| MarshallingLatency             | Timer         | The time taken to marshall the request
+| ApiCallAttemptCount            | Counter       | Total number of attempts that were made by the service client to fulfill this request before succeeding or failing. (Value is 1 if there are no retries)
```

Comment:
Not sure I follow the ask here. Is this not what this metric is?

Reply:
I mean the configured maximum number of attempts before it will give up. This metric shows the number of attempts that are actually used. Having both allows us to see how close it is to failing outright because it will exhaust all attempts.